R/ReactantExtra/build_tarballs.jl

```diff
@@ -0,0 +1,222 @@
+fi
+
+if [[ "${bb_full_target}" == *i686* ]]; then
+    BAZEL_BUILD_FLAGS+=(--define=build_with_mkl=false --define=enable_mkl=false)
+fi
+
+# julia --project=. -e "using Pkg; Pkg.instantiate(); Pkg.add(url=\"https://github.com/JuliaInterop/Clang.jl\", rev=\"vc/cxx_parse2\")"
+BAZEL_BUILD_FLAGS+=(--action_env=JULIA=julia)
+bazel ${BAZEL_FLAGS[@]} build ${BAZEL_BUILD_FLAGS[@]} ...
+rm bazel-bin/libReactantExtraLib*
+mkdir -p ${libdir}
+cp -v bazel-bin/libReactantExtra* ${libdir}
+cp bazel-bin/*.jl ${prefix}
+"""
+
+augment_platform_block = ""
+
+# determine exactly which tarballs we should build
+builds = []
+    
+# Dependencies that must be installed before this package can be built
+
+dependencies = []
+
+# The products that we will ensure are always built
+products = Product[
+    LibraryProduct(["libReactantExtra", "libReactantExtra"],
+                   :libReactantExtra, dont_dlopen=true),
+    FileProduct("Affine.inc.jl", Symbol("Affine.inc.jl")),
+    FileProduct("Arith.inc.jl", Symbol("Arith.inc.jl")),
+    FileProduct("Builtin.inc.jl", Symbol("Builtin.inc.jl")),
+    FileProduct("Enzyme.inc.jl", Symbol("Enzyme.inc.jl")),
+    FileProduct("Func.inc.jl", Symbol("Func.inc.jl")),
+    FileProduct("StableHLO.inc.jl", Symbol("StableHLO.inc.jl")),
+    FileProduct("CHLO.inc.jl", Symbol("CHLO.inc.jl")),
+    FileProduct("VHLO.inc.jl", Symbol("VHLO.inc.jl")),
+    # FileProduct("libMLIR_h.jl", Symbol("libMLIR_h.jl")),
+]
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = expand_cxxstring_abis(supported_platforms())
+
+# Don't even bother with powerpc
+platforms = filter(platforms) do p
+    !( (arch(p) == "powerpc64le") )
+end
+
+# 64-bit or bust (for xla runtime executable)
+platforms = filter(p -> arch(p) != "i686", platforms)
+
+# go sdk has weird issue for arm linux, removing for now
+platforms = filter(p -> !(arch(p) == "aarch64" && Sys.islinux(p)), platforms)
+platforms = filter(p -> !(arch(p) == "armv6l" && Sys.islinux(p)), platforms)
+platforms = filter(p -> !(arch(p) == "armv7l" && Sys.islinux(p)), platforms)
+
+# XLA has weird issue on musl
+# [22:07:03] bazel-out/k8-opt/bin/external/xla/xla/autotuning.pb.h:249:11: error: expected unqualified-id before ‘unsigned’
+# [22:07:03]   249 |   int32_t major() const;
+platforms = filter(p -> !(libc(p) == "musl"), platforms)
+
+# Windows has a cuda configure issue, to investigate either fixing/disabling cuda
+platforms = filter(p -> !(Sys.iswindows(p)), platforms)
+
+# NSync is picking up wrong stuff for cross compile, to deal with later
+# 02] ./external/nsync//platform/c++11.futex/platform.h:24:10: fatal error: 'linux/futex.h' file not found
+# [00:20:02] #include <linux/futex.h>
+platforms = filter(p -> !(Sys.isfreebsd(p)), platforms)
+platforms = filter(p -> !(Sys.isapple(p)), platforms)
+
+
+for platform in platforms
+    augmented_platform = deepcopy(platform)
```

Comment:
You aren't really augmenting the platforms, are you?  Do you plan to do it at a later stage (llvm version?  julia version?  anything else?)?  If not, this can be simplified.

Reply:
Eventually likely, but can change now if desired.